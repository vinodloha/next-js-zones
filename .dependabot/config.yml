# Dependabot is a service that automates the
# process of keeping your dependencies up
# to date with the latest security and bug
# fixes. To enable it for this repo, visit
# https://dependabot.com
version: 1
update_configs:
  - package_manager: "javascript"
    directory: "/"
    update_schedule: "daily"
    commit_message:
      prefix: "fix"
      prefix_development: "chore"
      include_scope: true
