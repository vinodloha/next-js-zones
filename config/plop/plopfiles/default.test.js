import config from './default'

const mockPlop = {}
const mockHelpers = {}
mockPlop.setGenerator = jest.fn()
mockPlop.addHelper = jest.fn((name, fn) => {
  mockHelpers[name] = fn
})

describe('plopfile:default', () => {
  afterEach(() => {
    mockPlop.setGenerator.mockClear()
    mockPlop.addHelper.mockClear()
  });
  it('provides the default plop configuration', () => {
    config(mockPlop)
    expect(mockPlop.setGenerator.mock.calls.length).toEqual(2)
    expect(mockPlop.setGenerator.mock.calls[0]).toMatchSnapshot()
  })
  describe('helpers.relativeSegments()', () => {
    it('generates a relative path with depth matching provided depth', () => {
      config(mockPlop)
      expect(mockPlop.setGenerator.mock.calls.length).toEqual(2)
      expect(mockHelpers.relativeSegments('my/example/page')).toEqual('../../')
      expect(mockHelpers.relativeSegments('page')).toEqual('')
    })
  })
})
