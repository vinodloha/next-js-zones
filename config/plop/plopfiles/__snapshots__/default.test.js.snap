// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plopfile:default provides the default plop configuration 1`] = `
Array [
  "component",
  Object {
    "actions": Array [
      Object {
        "path": "src/components/{{atomicType}}s/{{properCase componentName}}/index.js",
        "templateFile": "../templates/components/{{atomicType}}/index.js.hbs",
        "type": "add",
      },
      Object {
        "path": "src/components/{{atomicType}}s/{{properCase componentName}}/{{properCase componentName}}.jsx",
        "templateFile": "../templates/components/{{atomicType}}/{{properCase atomicType}}.jsx.hbs",
        "type": "add",
      },
      Object {
        "path": "src/components/{{atomicType}}s/{{properCase componentName}}/{{properCase componentName}}.test.jsx",
        "templateFile": "../templates/components/{{atomicType}}/{{properCase atomicType}}.test.jsx.hbs",
        "type": "add",
      },
    ],
    "description": "Create a component",
    "prompts": Array [
      Object {
        "choices": Array [
          "atom",
          "molecule",
          "organism",
        ],
        "message": "What level is this component?",
        "name": "atomicType",
        "type": "list",
      },
      Object {
        "message": "Name of component?",
        "name": "componentName",
        "type": "input",
      },
      Object {
        "message": "Short description of the component (eg: \`a list of names\`):",
        "name": "componentDescription",
        "type": "input",
      },
    ],
  },
]
`;
